cscs-checks/prgenv/environ_check.py

```diff
@@ -58,58 +58,62 @@ def set_sanity(self):
 
-@rfm.parameterized_test(['cray-fftw'], ['cray-hdf5'], ['cray-hdf5-parallel'],
-                        ['cray-libsci'], ['cray-mpich'], ['cray-netcdf'],
-                        ['cray-netcdf-hdf5parallel'], ['cray-petsc'],
-                        ['cray-petsc-complex-64'], ['cray-python'],
-                        ['cray-R'], ['cray-tpsl'], ['cray-tpsl-64'],
-                        ['cudatoolkit'], ['gcc'], ['papi'], ['pmi'])
-class CrayVariablesCheck(rfm.RunOnlyRegressionTest, CrayVariablesCheckMixin):
-    def __init__(self, module_name):
-        CrayVariablesCheckMixin.__init__(self, module_name)
+@rfm.simple_test
+class CrayVariablesCheck(CrayVariablesCheckBase):
+    cray_module = parameter([
+        'cray-fftw', 'cray-hdf5', 'cray-hdf5-parallel', 'cray-libsci',
+        'cray-mpich', 'cray-netcdf', 'cray-netcdf-hdf5parallel', 'cray-petsc',
+        'cray-petsc-complex-64', 'cray-python', 'cray-R', 'cray-tpsl',
+        'cray-tpsl-64', 'cudatoolkit', 'gcc', 'papi', 'pmi'
+    ])
+
+    def __init__(self):
+        super().__init__()
         self.valid_systems = ['daint:login', 'dom:login']
 
         # FIXME: These modules should be fixed in later releases,
         # while gcc was fixed in 20.11
 
         cdt = osext.cray_cdt_version()
         if ((cdt and cdt <= '20.11' and
-             module_name in ['cray-petsc-complex',
-                             'cray-petsc-complex-64',
-                             'cudatoolkit']) or
+             self.cray_module in ['cray-petsc-complex',
+                                  'cray-petsc-complex-64',
+                                  'cudatoolkit']) or
             (cdt and cdt < '20.11' and module_name == 'gcc')):
             self.valid_systems = []
 
         self.maintainers = ['EK', 'VH']
 
 
-@rfm.parameterized_test(['cray-fftw'], ['cray-hdf5'], ['cray-hdf5-parallel'],
-                        ['cray-libsci'], ['cray-mpich'], ['cray-openshmemx'],
-                        ['cray-parallel-netcdf'], ['cray-pmi'],
-                        ['cray-python'], ['cray-R'], ['gcc'], ['papi'])
-class CrayVariablesCheckEiger(rfm.RunOnlyRegressionTest,
-                              CrayVariablesCheckMixin):
-    def __init__(self, module_name):
-        CrayVariablesCheckMixin.__init__(self, module_name)
+@rfm.simple_test
+class CrayVariablesCheckEiger(CrayVariablesCheckBase):
+    cray_module = parameter([
+        'cray-fftw', 'cray-hdf5', 'cray-hdf5-parallel', 'cray-libsci',
+        'cray-mpich', 'cray-openshmemx', 'cray-parallel-netcdf', 'cray-pmi',
+        'cray-python', 'cray-R', 'gcc', 'papi'
+    ])
+
+    def __init__(self):
+        super().__init__()
         self.valid_systems = ['eiger:login']
 
         # FIXME: These modules should be fixed in later releases
 
-        if module_name in {'cray-fftw', 'cray-python', 'cray-mpich'}:
+        if self.cray_module in {'cray-fftw', 'cray-python', 'cray-mpich'}:
             self.valid_systems = []
 
         self.maintainers = ['TM']
```

Comment:
Why don't you put the same maintainers in both tests? You may add you and @ekouts.